### PR TITLE
Add DataStax Enterprise as a Cassandra test environment

### LINKED ISSUE #10444: Add DataStax as Cassandra test environment
As far as I confirmed, `5.1.5` works without any changes, but `6.8.18` fails to start a query runner due to below messages:
```bash
2022-01-01T14:37:44.296+0900	ERROR	main	Bootstrap	Uncaught exception in thread main
java.lang.RuntimeException: Schema tpch not found
	at io.trino.testing.AbstractTestingTrinoClient.execute(AbstractTestingTrinoClient.java:122)
	at io.trino.testing.DistributedQueryRunner.execute(DistributedQueryRunner.java:505)
	at io.trino.testing.QueryAssertions.copyTable(QueryAssertions.java:339)
	at io.trino.testing.QueryAssertions.copyTable(QueryAssertions.java:331)
	at io.trino.testing.QueryAssertions.copyTpchTables(QueryAssertions.java:323)
	at io.trino.plugin.cassandra.DataStaxQueryRunner.createScyllaQueryRunner(DataStaxQueryRunner.java:61)
	at io.trino.plugin.cassandra.DataStaxQueryRunner.main(DataStaxQueryRunner.java:86)
	Suppressed: java.lang.Exception: SQL: CREATE TABLE IF NOT EXISTS customer AS SELECT * FROM tpch.tiny.customer
		at io.trino.testing.DistributedQueryRunner.execute(DistributedQueryRunner.java:508)
		... 5 more
Caused by: io.trino.spi.connector.SchemaNotFoundException: Schema tpch not found
	at io.trino.plugin.cassandra.CassandraSession.getKeyspaceByCaseInsensitiveName(CassandraSession.java:274)
	at io.trino.plugin.cassandra.CassandraSession.getCaseSensitiveSchemaName(CassandraSession.java:154)
	at io.trino.plugin.cassandra.CassandraMetadata.createTable(CassandraMetadata.java:294)
	at io.trino.plugin.cassandra.CassandraMetadata.beginCreateTable(CassandraMetadata.java:278)
	at io.trino.spi.connector.ConnectorMetadata.beginCreateTable(ConnectorMetadata.java:528)
	at io.trino.metadata.MetadataManager.beginCreateTable(MetadataManager.java:955)
	at io.trino.sql.planner.optimizations.BeginTableWrite$Rewriter.createWriterTarget(BeginTableWrite.java:253)
	at io.trino.sql.planner.optimizations.BeginTableWrite$Rewriter.visitTableFinish(BeginTableWrite.java:197)
	at io.trino.sql.planner.optimizations.BeginTableWrite$Rewriter.visitTableFinish(BeginTableWrite.java:100)
	at io.trino.sql.planner.plan.TableFinishNode.accept(TableFinishNode.java:106)
	at io.trino.sql.planner.plan.SimplePlanRewriter$RewriteContext.rewrite(SimplePlanRewriter.java:84)
	at io.trino.sql.planner.plan.SimplePlanRewriter$RewriteContext.lambda$defaultRewrite$0(SimplePlanRewriter.java:73)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.Collections$2.tryAdvance(Collections.java:4747)
	at java.base/java.util.Collections$2.forEachRemaining(Collections.java:4755)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
	at io.trino.sql.planner.plan.SimplePlanRewriter$RewriteContext.defaultRewrite(SimplePlanRewriter.java:74)
	at io.trino.sql.planner.plan.SimplePlanRewriter.visitPlan(SimplePlanRewriter.java:38)
	at io.trino.sql.planner.plan.SimplePlanRewriter.visitPlan(SimplePlanRewriter.java:22)
	at io.trino.sql.planner.plan.PlanVisitor.visitOutput(PlanVisitor.java:49)
	at io.trino.sql.planner.plan.OutputNode.accept(OutputNode.java:83)
	at io.trino.sql.planner.plan.SimplePlanRewriter.rewriteWith(SimplePlanRewriter.java:32)
	at io.trino.sql.planner.optimizations.BeginTableWrite.optimize(BeginTableWrite.java:85)
	at io.trino.sql.planner.LogicalPlanner.plan(LogicalPlanner.java:230)
	at io.trino.sql.planner.LogicalPlanner.plan(LogicalPlanner.java:215)
	at io.trino.sql.planner.LogicalPlanner.plan(LogicalPlanner.java:210)
	at io.trino.execution.SqlQueryExecution.doPlanQuery(SqlQueryExecution.java:466)
	at io.trino.execution.SqlQueryExecution.planQuery(SqlQueryExecution.java:447)
	at io.trino.execution.SqlQueryExecution.start(SqlQueryExecution.java:388)
	at io.trino.execution.SqlQueryManager.createQuery(SqlQueryManager.java:243)
	at io.trino.dispatcher.LocalDispatchQuery.lambda$startExecution$7(LocalDispatchQuery.java:143)
	at io.trino.$gen.Trino_testversion____20220101_053741_3.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
	Suppressed: java.lang.Exception: Current plan:
                Output[rows]
                │   Layout: [rows:bigint]
                └─ TableCommit[cassandra.tpch.customer]
                   │   Layout: [rows:bigint]
                   └─ LocalExchange[SINGLE] ()
                      │   Layout: [partialrows:bigint, fragment:varbinary]
                      └─ RemoteExchange[GATHER]
                         │   Layout: [partialrows:bigint, fragment:varbinary]
                         └─ TableWriter
                            │   Layout: [partialrows:bigint, fragment:varbinary]
                            │   custkey := custkey
                            │   name := name
                            │   address := address
                            │   nationkey := nationkey
                            │   phone := phone
                            │   acctbal := acctbal
                            │   mktsegment := mktsegment
                            │   comment := comment
                            └─ LocalExchange[SINGLE] ()
                               │   Layout: [custkey:bigint, name:varchar(25), address:varchar(40), nationkey:bigint, phone:varchar(15), acctbal:double, mktsegment:varchar(10), comment:varchar(117)]
                               └─ RemoteExchange[REPARTITION]
                                  │   Layout: [custkey:bigint, name:varchar(25), address:varchar(40), nationkey:bigint, phone:varchar(15), acctbal:double, mktsegment:varchar(10), comment:varchar(117)]
                                  └─ TableScan[tpch:tiny:customer]
                                         Layout: [custkey:bigint, name:varchar(25), address:varchar(40), nationkey:bigint, phone:varchar(15), acctbal:double, mktsegment:varchar(10), comment:varchar(117)]
                                         nationkey := tpch:nationkey
                                         mktsegment := tpch:mktsegment
                                         address := tpch:address
                                         phone := tpch:phone
                                         custkey := tpch:custkey
                                         name := tpch:name
                                         comment := tpch:comment
                                         acctbal := tpch:acctbal

		at io.trino.sql.planner.optimizations.BeginTableWrite.optimize(BeginTableWrite.java:91)
		... 12 more
```

https://hub.docker.com/r/datastax/dse-server

### PR DESCRIPTION
**Pull request by kamil-rafalko**
## Description

This PR adds a new test environment to the Cassandra connector smoke tests in particular DataStax Enterprise in version 6.8.25 (for more information please visit [docs](https://docs.datastax.com/en/docker/doc/index.html?utm_campaign=Docker_Cus_2019&utm_medium=web&utm_source=docker&utm_term=&utm_content=Web_DocsDocker))

## Related issues, pull requests, and links
Closes #10444

## Documentation

(x) No documentation is needed.
( ) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

(x) No release notes entries required.
( ) Release notes entries required with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/CassandraServer.java [hunk 1]
```diff
@@ -57,27 +59,39 @@
 {
     private static Logger log = Logger.get(CassandraServer.class);
 
-    private static final int PORT = 9142;
+    private static final int DEFAULT_PORT = 9142;
```
Comment (ebyhr) [inline plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/CassandraServer.java:62]: Can you try changing `native_transport_port` value to 9042 in `cu-cassandra.yaml` in a preparatory commit? Thus, we don't need to add `int port` argument to the constructor.

File: plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/CassandraServer.java [hunk 2]
```diff
@@ -57,27 +59,39 @@
...
 
     private final GenericContainer<?> dockerContainer;
     private final CassandraSession session;
+    private final int port;
 
     public CassandraServer()
             throws Exception
     {
-        this("2.2");
+        this("cassandra", "2.2", DEFAULT_PORT, Collections.emptyMap());
```
Comment (ebyhr) [inline plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/CassandraServer.java:73]: Could you please use `ImmutableMap.of()`?

File: plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/CassandraServer.java [hunk 3]
```diff
@@ -57,27 +59,39 @@
...
     }
 
     public CassandraServer(String cassandraVersion)
             throws Exception
+    {
+        this("cassandra", cassandraVersion, DEFAULT_PORT, Collections.emptyMap());
```
Comment (ebyhr) [inline plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/CassandraServer.java:79]: ditto.

File: plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/CassandraServer.java [hunk 4]
```diff
@@ -57,27 +59,39 @@
...
+
+    public CassandraServer(String imageName, String cassandraVersion, int port, Map<String, String> additionalEnvVariables)
+            throws Exception
     {
         log.info("Starting cassandra...");
 
-        this.dockerContainer = new GenericContainer<>("cassandra:" + cassandraVersion)
-                .withExposedPorts(PORT)
-                .withCopyFileToContainer(forHostPath(prepareCassandraYaml()), "/etc/cassandra/cassandra.yaml");
+        this.dockerContainer = additionalEnvVariables.entrySet().stream()
```
Comment (ebyhr) [inline plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/CassandraServer.java:87]: Why not calling `withEnv(Map<String, String> env)`?

File: plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/TestDatastaxConnectorSmokeTest.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+public class TestDatastaxConnectorSmokeTest
+        extends BaseCassandraConnectorSmokeTest
+{
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        CassandraServer server = closeAfterClass(new CassandraServer(
+                "datastax/dse-server",
+                "6.8.25",
```
Comment (findepi) [inline plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/TestDatastaxConnectorSmokeTest.java:34]: nit 
Imo it's better if the two come together as a single String

```suggestion
                "datastax/dse-server:6.8.25",
```

File: plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/TestDatastaxConnectorSmokeTest.java [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+                9042,
+                Map.of(
+                        "DS_LICENSE", "accept",
```
Comment (findepi) [inline plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/TestDatastaxConnectorSmokeTest.java:37]: > Set to accept to acknowledge that you agree with the terms of the DataStax license.

What are the license conditions?

Comment (kamil-rafalko) [reply plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/TestDatastaxConnectorSmokeTest.java:37]: I think that part of the license that we should be interested in is here https://www.datastax.com/legal/datastax-developer-license-terms. I think that there is nothing that blocks us from using it, but I'm not an expert in terms of licenses.

File: plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/CassandraServer.java [hunk 7]
```diff
@@ -67,17 +69,25 @@
...
     }
 
-    public CassandraServer(String cassandraVersion)
+    public CassandraServer(String imageName)
+            throws Exception
+    {
+        this(imageName, ImmutableMap.of());
+    }
+
+    public CassandraServer(String imageName, Map<String, String> additionalEnvVariables)
```
Comment (ebyhr) [inline plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/CassandraServer.java:81]: We avoid abbreviations as much as possible. Let's rename to `environmentVariables`.

File: plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/CassandraServer.java [hunk 8]
```diff
@@ -67,17 +69,25 @@
...
             throws Exception
     {
         log.info("Starting cassandra...");
 
-        this.dockerContainer = new GenericContainer<>("cassandra:" + cassandraVersion)
+        this.dockerContainer = new GenericContainer<>(imageName)
                 .withExposedPorts(PORT)
-                .withCopyFileToContainer(forHostPath(prepareCassandraYaml()), "/etc/cassandra/cassandra.yaml");
+                .withCopyFileToContainer(forHostPath(prepareCassandraYaml()), "/etc/cassandra/cassandra.yaml")
+                .withEnv(additionalEnvVariables);
```
Comment (ebyhr) [inline plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/CassandraServer.java:89]: The default startup timeout looks insufficient. Could you add a below line?
```
.withStartupTimeout(java.time.Duration.ofMinutes(10))
```

File: plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/CassandraServer.java [hunk 9]
```diff
@@ -67,17 +69,25 @@
...
+
```
Comment (ebyhr) [inline plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/CassandraServer.java:90]: Revert unrelated change (= remove this empty line)

**Review by ebyhr — COMMENTED** [review]
Could you please reorder the commit as
1. Merge docker image name and tag in Cassandra
2. Change Cassandra port in cu-casandra.yaml 
3. Add DataStax Enterprise as a Cassandra test environment

The preparatory commits should be first.

**Review by ebyhr — COMMENTED** [review]
I ran stress tests in #13623 with these commits (red) and #13625 without these commits (green) because adding a new test environment sometimes leads to flaky tests. Can you debug why #13623 failed?

**Comment by kamil-rafalko** [comment]
@ebyhr I think I've managed to fix unstable tests. It turned out that path that we currently use as a `cassandra.yaml` path is not valid in Datastax Enterprise image. I had to parameterize this directory in order to provide a different path for DSE. Additionally, current config file is not 100% compatible with DSE so I've prepared an alternative, minimal version of it. Please review it one more time.

**Review by ebyhr — APPROVED** [review]

**Comment by ebyhr** [comment]
Merged, thanks!
